src/api/hooks.cc

```diff
@@ -196,6 +196,12 @@ async_id AsyncHooksGetExecutionAsyncId(Isolate* isolate) {
   return env->execution_async_id();
 }
 
+async_id AsyncHooksGetExecutionAsyncId(Local<Context> context) {
+  Environment* env = Environment::GetCurrent(context);
+  if (env == nullptr) return -1;
```

Comment:
Not for this PR but we might consider changing these to use `std::optional<async_id>` as the return value rather than -1.

Reply:
Actually... ugh, nevermind. I remember these are part of the `node.h` exposed API so we can't change the existing ones without breaking ABI. Sigh.